pennylane/capture/capture_qnode.py

```diff
@@ -78,9 +82,85 @@ def qfunc(*inner_args):
 
     # pylint: disable=unused-argument
     @qnode_prim.def_abstract_eval
-    def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts):
+    def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts, batch_shape=()):
+
         mps = qfunc_jaxpr.outvars
-        return _get_shapes_for(*mps, shots=shots, num_device_wires=len(device.wires))
+
+        return _get_shapes_for(
+            *mps,
+            shots=shots,
+            num_device_wires=len(device.wires),
+            batch_shape=batch_shape,
+        )
+
+    # pylint: disable=too-many-arguments, too-many-positional-arguments
+    def _qnode_batching_rule(
+        batched_args,
+        batch_dims,
+        qnode,
+        shots,
+        device,
+        qnode_kwargs,
+        qfunc_jaxpr,
+        n_consts,
+    ):
+        """
+        Batching rule for the ``qnode`` primitive.
+
+        This rule exploits the parameter broadcasting feature of the QNode to vectorize the circuit execution.
+        """
+
+        assert len(batched_args) == len(batch_dims), "Mismatch in batched arguments and dimensions."
+        assert all(
+            batch_dim is None or isinstance(batch_dim, int) for batch_dim in batch_dims
+        ), "Invalid batch dimensions found."
+
+        args = batched_args[n_consts:]
+
+        for i, (arg, batch_dim) in enumerate(zip(batched_args, batch_dims)):
+
+            if not is_non_scalar_tensor(arg):
+                continue
```

Comment:
Could we perhaps have a private `_is_scalar_tensor` so our brains to have to handle a double negative?